### ISSUE TITLE
[73X] Update LLVM/Clang

### ISSUE THREAD

**Issue by davidlt** [cms-sw/cmsdist#1174]
`RecoEgamma/EgammaIsolationAlgos` package is ICE'ing the Clang and `RecoVertex/AdaptiveVertexFit` is capturing non-POD VLAs in lambda. The latter [landed in Clang](http://reviews.llvm.org/D4368) after 3.5.0 release.

Both of these are resolved in newer Clang versions.

The following seems to resolve it.

```
diff --git a/llvm.spec b/llvm.spec
index dc1532e..e44600a 100644
--- a/llvm.spec
+++ b/llvm.spec
@@ -4,11 +4,12 @@
 BuildRequires: python
 Requires: gcc

-%define llvmCommit f55c17bc339
-%define clangCommit 656aa83af0b
-%define branch cms/v%realversion
-Source0: git+https://github.com/cms-externals/llvm.git?obj=%{branch}/%{llvmCommit}&export=llvm-%{realversion}-%{llvmCommit}&module=llvm-%realversion-%llvmCommit&output=/llvm-%{real
-Source1: git+https://github.com/cms-externals/clang.git?obj=%{branch}/%{clangCommit}&export=clang-%{realversion}-%{clangCommit}&module=clang-%realversion-%clangCommit&output=/clang
+%define llvmCommit e1e1392e13ef3f5e97fbdd15ca8d60d63f69a862
+%define clangCommit 097068701d2c7fff58866059b1e3ebcac2728f7a
+%define llvmBranch cms/e1e1392
+%define clangBranch cms/9443d04
+Source0: git+https://github.com/davidlt/llvm.git?obj=%{llvmBranch}/%{llvmCommit}&export=llvm-%{realversion}-%{llvmCommit}&module=llvm-%realversion-%llvmCommit&output=/llvm-%{realve
+Source1: git+https://github.com/davidlt/clang.git?obj=%{clangBranch}/%{clangCommit}&export=clang-%{realversion}-%{clangCommit}&module=clang-%realversion-%clangCommit&output=/clang-

 # Still need forward porting.
 %define keep_archives true
```

LLVM upstream commit `e1e1392e13ef3f5e97fbdd15ca8d60d63f69a862`
Clang upstream commit `9443d047e89e716928fd87d277fca673afc2ec43`

This is a working item for me to find and propose working LLVM and Clang commits for the update.

**Comment by ktf** [comment]
+1

**Comment by davidlt** [comment]
I doubt there is a working set of commits currently. I started reporting digging into and reporting the issues to get them fixed for trunk/3.6.

This issue will track a progress.

http://llvm.org/bugs/show_bug.cgi?id=21403 is affecting `RecoPixelVertexing/PixelTriplets` 

The patch not tested by me, available here: http://reviews.llvm.org/D6124 (still under review)

**Comment by ktf** [comment]
Any progress on this? Otherwise please close.

**Comment by davidlt** [comment]
I have the commits, but will wait until you get root/xrootd/boost from 7_3_DEVEL_X to 7_4_X. Then I will add it to 7_3_DEVEL_X for a cross-check before suggesting to bump LLVM/Clang in 7_4_X.

**Comment by ktf** [comment]
As discussed, let's do this in CMSSW_7_4_DEVEL_X, once opened.

**Comment by ktf** [comment]
What's the status of this one?

**Comment by davidlt** [comment]
The new LLVM/Clang is in DEVEL_X for the last ~6 weeks I believe, but compiling of CMSSW with Clang was completely broken after we added some thread safety patches in ROOT. I sent RFC patch to Philippe, which was a double fix. He did commit it and Chris recently requested it backported into our own fork, which I believe was done recently.

After that I haven't continued looking in CMSSW  + Clang, but I should as 3.6 is close to be released.

Closing as goal of PR is achieved.
